### PR TITLE
WIP - Integrate with InfluxDB instead of KeenIO

### PR DESCRIPTION
**Pull request by soloman1124**

### CODE REVIEW

File: lib/nark.rb [hunk 1]
```diff
@@ -24,4 +25,23 @@ def collection_name(value = nil)
...
+    end
+
+    def emitter
+      @emitter ||= InfluxDBEmitter.new(
+        ENV['INFLUXDB_DATABASE'],
+        username: ENV['INFLUXDB_USERNAME'],
+        password: ENV['INFLUXDB_PASSWORD'],
+        hosts: ENV['INFLUXDB_HOST'],
+        port: ENV['INFLUXDB_PORT'],
+        use_ssl: ENV['INFLUXDB_USE_SSL']
```
Comment (coop) [inline lib/nark.rb:60]: I really wish there was an easier way to do this mapping (`ENV -> config`). At one stage I think @JonathonMA wrote a library to handle it but it was a bit more code than manually typing it out like you've done here. Just for shits and giggles I think the following would work but it definitely feels more magical (I'd stick with what you have).

``` ruby
ENV
  .select { |key, _value| key.starts_with?("INFLUXDB_") }
  .each_with_object({}) { |(key, value), config| config[key.replace("INFLUXDB_").downcase] = value }
```

Comment (coop) [inline lib/nark.rb:60]: Useless ruby tip... Carry on.

Comment (soloman1124) [inline lib/nark.rb:60]: @coop, thanks for the suggestion~ yeah, I also think the one you have suggested might require some documentation comments to find out what environment variables are required to be set.

:smile:

Comment (coop) [inline lib/nark.rb:60]: You're right.

Comment (JonathonMA) [inline lib/nark.rb:60]: I'm pretty sure @coop is thinking of the `righteously_slice` method. It doesn't exist. :(

Comment (evilmarty) [inline lib/nark.rb:60]: Needs more...

**Pull request merged**